### PR TITLE
Added documentation for NicaDrone EPM V3

### PR DESCRIPTION
**Pull request by ctech4285**

### CODE REVIEW

**Comment by pavel-kirienko** [comment]
+1 this is a good update

File: common/source/docs/common-electro-permanent-magnet-V3.rst [hunk 1]
```diff
@@ -0,0 +1,143 @@
+.. _common-electro-permanent-magnet-gripper:
+
+=========================================
```
Comment (hamishwillee) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: These need to extend to length of title

File: common/source/docs/common-electro-permanent-magnet-V3.rst [hunk 2]
```diff
@@ -0,0 +1,143 @@
+.. _common-electro-permanent-magnet-gripper:
```
Comment (hamishwillee) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: This is the same anchor as used by [our other emp topic](https://github.com/ArduPilot/ardupilot_wiki/blob/master/common/source/docs/common-electro-permanent-magnet-gripper.rst). Propose you rename the FILE to common-electro-permanent-magnet-nicodrone-emp-v3.rst and the anchor to  `_common-electro-permanent-magnet-gripper_nicodrone_epm_v3`

File: common/source/docs/common-electro-permanent-magnet-V3.rst [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+=========================================
+Electro Permanent Magnet Gripper (OpenGrab EPM v3)
+=========================================
+
+This article shows how to use the `Electro Permanent Magnet Gripper (OpenGrab EPM v3) <http://nicadrone.com/index.php?id_product=66&controller=product>`__
+with ArduPilot.
+
+`Problems issues feedback <mailto:Andreas@NicaDrone.com?Subject=Problems%20issues%20feedback>`__
```
Comment (hamishwillee) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: "Problems/issues"

File: common/source/docs/common-electro-permanent-magnet-V3.rst [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+
+Overview
+========
+
+The *OpenGrab EPM v3* (from NicaDrone.com) is a small (less than 4cm x 4cm x 2cm)
+Electro Permanent Magnet gripper capable of 20kg of force meaning it can
```
Comment (hamishwillee) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: kg are a unit of mass, not force. Please clarify what you're trying to say here. I would suggest perhaps "gripper capable of lifting about 15kg of cargo" would be useful.

File: common/source/docs/common-electro-permanent-magnet-V3.rst [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+hold about 15kg of cargo.  The gripping and releasing of the cargo if
```
Comment (hamishwillee) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: "if accomplished"?

File: common/source/docs/common-electro-permanent-magnet-V3.rst [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+the included `thin steel target square <http://nicadrone.com/index.php?id_product=15&controller=product>`__
+attached to them.
+
+The device accepts a standard PWM servo input or UAVCAN and uses a 5V power
+supply.  Significant amounts of power are only required for about about 1
+second when it first grips or releases the cargo.
+
+`See demo **Video** of the product here <https://youtu.be/ggvm-GQxwaY>`__!
+
+Usefull links
```
Comment (hamishwillee) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: Spelling "useful"

File: common/source/docs/common-electro-permanent-magnet-V3.rst [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+- The Pixhwak power rail voltage is <<5V
+- External powering of the EPM is recommended see image above
+- EPM UAVCAN cable set is required. `Clickmate -> DF13 <http://nicadrone.com/index.php?id_product=69&controller=product>`__
+- UAVCAN requires termination, terminator included in UAVCAN cable set
+
+
+Firmware setup
+==============
+
+1.       Download APM firmware `here: <https://files.zubax.com/3rdparty/APM/uavcan_epm/>`__
```
Comment (hamishwillee) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: Auto numbering is better. Do as:

```
#. Download ...
#. Upload ...
```

File: common/source/docs/common-electro-permanent-magnet-V3.rst [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+2.       Upload custom firmware via Mission Planner or QGC
+ 
+***A pull request has been issued to include the changes in the next APM release. This documentation will be update when the request has been accepted. If this annoys you please write a comment `here <https://github.com/ArduPilot/ardupilot/pull/4361>`__
+
+Parameter setup
+===============
+
+1.       Set BRD_CAN_ENABLE = 2 ( Enabled support for dynamic node ID allocation and firmware update.)
```
Comment (hamishwillee) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: Auto numbering is better here, but you will also have to do indentation to get the layout right. So something like:

```
#. Set BRD_CAN_ENABLE = 2 ( Enabled support for dynamic node ID allocation and firmware update.)
   Save and restart the board. 
   After about 5 seconds the RED CAN LED on the EPM will start blinking. Indicating that CAN frames are being received
   
   .. image:: ../../../images/OpenGrabEPMV3_3.jpg
       :target: ../_images/OpenGrabEPMV3_3.jpg

 #. Set CH7_OPT: Channel 7 option = 19 (EPM) (Channel 7 - 12 maybe used) 
```
Note that above each line is indented to the bullet point (3 spaces). This then puts the image "inside" the bullet.

File: common/source/docs/common-electro-permanent-magnet-V3.rst [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+===============
+
+
+Set CH7_OPT: Channel 7 option = 19 (EPM) (Channel 7 - 12 maybe used) 
+
+.. image:: ../../../images/OpenGrabEPMV3_6.jpg
+    :target: ../_images/OpenGrabEPMV3_6.jpg
+    
+    
+All done, The EPM can be controlled via channal 7 or auto Mission or MAVLink command
```
Comment (hamishwillee) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: Finish sentences with full stop.

File: common/source/docs/common-electro-permanent-magnet-V3.rst [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+.. image:: ../../../images/OpenGrabEPMV3_6.jpg
+    :target: ../_images/OpenGrabEPMV3_6.jpg
+    
+    
+All done, The EPM can be controlled via channal 7 or auto Mission or MAVLink command     
+
+
+
+
+Auto Mission
```
Comment (hamishwillee) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: use sentence capitalisation on page headings (e.g. Auto mission)

**Comment by hamishwillee** [comment]
This looks pretty useful to me but needs a good subedit and some layout fixes. Minimally file name and main anchor need to be fixed.

This will also need to be added into the hierarchy, which means a [modification of the parent page](https://github.com/ArduPilot/ardupilot_wiki/blob/master/common/source/docs/common-optional-hardware.rst). See the end of the page at the line:
```
EPM Gripper (EPM688) <common-electro-permanent-magnet-gripper>
```
Typically we'd create a parent grouping page now there are two EPMs, but for now just add to this page.

@rmackay9 @magicrub - Other than the obvious, are you OK with this going in?

**Comment by rmackay9** [comment]
I'm certainly happy with more documentation on the EPM being added to the wiki.

File: common/source/docs/common-electro-permanent-magnet-V3.rst (hunk 1 above)
Comment (ctech4285) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: done

File: common/source/docs/common-electro-permanent-magnet-V3.rst (hunk 3 above)
Comment (ctech4285) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: done

File: common/source/docs/common-electro-permanent-magnet-V3.rst (hunk 4 above)
Comment (ctech4285) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: Done

File: common/source/docs/common-electro-permanent-magnet-V3.rst (hunk 5 above)
Comment (ctech4285) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: done

File: common/source/docs/common-electro-permanent-magnet-V3.rst (hunk 6 above)
Comment (ctech4285) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: done

File: common/source/docs/common-electro-permanent-magnet-V3.rst (hunk 7 above)
Comment (ctech4285) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: done

File: common/source/docs/common-electro-permanent-magnet-V3.rst (hunk 8 above)
Comment (ctech4285) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: done

File: common/source/docs/common-electro-permanent-magnet-V3.rst (hunk 9 above)
Comment (ctech4285) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: done

File: common/source/docs/common-electro-permanent-magnet-V3.rst (hunk 10 above)
Comment (ctech4285) [inline common/source/docs/common-electro-permanent-magnet-V3.rst]: done

**Comment by ctech4285** [comment]
Hello hamishwillee, thank you very much for your feedback. All changes have been done hopefully correctly

**Comment by hamishwillee** [comment]
Thanks @ctech4285 ! Changes good. I may do some further tidy, but that can happen as a post process. Merged!

**Comment by hamishwillee** [comment]
PS Fixes done in #506

**Pull request merged**